crates/red_knot_module_resolver/src/path.rs

```diff
@@ -197,6 +211,18 @@ impl ModuleResolutionPathBuf {
             .then_some(Self(ModuleResolutionPathBufInner::SitePackages(path)))
     }
 
+    #[must_use]
+    pub(crate) fn editable_installation_root(
+        db: &dyn Db,
+        path: impl Into<SystemPathBuf>,
+    ) -> Option<Self> {
+        let path = path.into();
+        // TODO: Add Salsa invalidation to this system call:
+        db.system()
+            .is_directory(&path)
```

Comment:
At runtime, Python does no validation for whether or not the path is a directory before adding it to `sys.path`: https://docs.python.org/3/library/site.html#module-site. Possibly we shouldn't either? I should check what mypy and pyright do here.

Reply:
Yeah, if you append a file to `sys.path` at runtime it doesn't seem to have any impact on runtime semantics.

Reply:
Putting a file on `sys.path` is meaningful at runtime if there is an import loader that understands such an entry. For instance, the builtin zipfile import loader can load modules from zip files listed on `sys.path`. The homebrew-installed `python3` on my MacBook actually has the path `.../lib/python312.zip` on `sys.path` by default.

I don't think it's totally out of the question that we'd support zip loader someday (it wouldn't be technically all that hard, since we're already doing it for vendored files), but it seems pretty unlikely, since these days it's not much used AFAIK. So I think it's fine to just assume/require only directories on `sys.path` for now.